server/controllers/files.ts

```diff
@@ -0,0 +1,109 @@
+import config from 'config';
+import { Request, Response } from 'express';
+
+import { canSeeExpenseAttachments } from '../graphql/common/expenses';
+import { getSignedGetURL, parseS3Url } from '../lib/awsS3';
+import { Collective, Expense, ExpenseAttachedFile, ExpenseItem, UploadedFile } from '../models';
+import { SUPPORTED_FILE_TYPES_IMAGES } from '../models/UploadedFile';
+
+async function hasUploadedFilePermission(req: Request, uploadedFile: UploadedFile): Promise<boolean> {
+  const actualUrl = uploadedFile.getDataValue('url');
+  if (req.remoteUser.id !== uploadedFile.CreatedByUserId) {
+    switch (uploadedFile.kind) {
+      case 'EXPENSE_ITEM': {
+        const expenseItem = await ExpenseItem.findOne({
+          where: {
+            url: actualUrl,
+          },
+          include: { model: Expense, include: [{ model: Collective, as: 'fromCollective' }] },
```

Comment:
:ice_cream: I tend to find the `association` parameter clearer when it's available

```suggestion
          include: { model: Expense, include: [{ association: 'fromCollective' }] },
```